# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1419141601 18000
#      Sun Dec 21 01:00:01 2014 -0500
# Node ID e354372e20f2360f6b0c1b3cc8d86fade1d5af19
# Parent  78e424d315967ed7690ab7f3d17d8c6df050866d
allow sparse arrays to be indexed by N-d arrays (bug #43855)

* Sparse.cc (Sparse<T>::index (const idx_vector&, bool)):
Smash 3rd and higher dimensions of idx vector to allow indexing by N-d
arrays.

diff --git a/liboctave/array/Sparse.cc b/liboctave/array/Sparse.cc
--- a/liboctave/array/Sparse.cc
+++ b/liboctave/array/Sparse.cc
@@ -1391,22 +1391,19 @@ Sparse<T>::index (const idx_vector& idx,
   assert (ndims () == 2);
 
   octave_idx_type nr = dim1 ();
   octave_idx_type nc = dim2 ();
   octave_idx_type nz = nnz ();
 
   octave_idx_type nel = numel (); // Can throw.
 
-  const dim_vector idx_dims = idx.orig_dimensions ();
-
-  if (idx_dims.length () > 2)
-    (*current_liboctave_error_handler)
-      ("cannot index sparse matrix with an N-D Array");
-  else if (idx.is_colon ())
+  const dim_vector idx_dims = idx.orig_dimensions ().redim (2);
+
+  if (idx.is_colon ())
     {
       if (nc == 1)
         retval = *this;
       else
         {
           // Fast magic colon processing.
           retval = Sparse<T> (nel, 1, nz);
 
